### PR TITLE
feat: post editing pages support caching of content to the browser

### PR DESCRIPTION
**Pull request by Aanko**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
支持将文章内容实时保存到浏览器，防止意外操作丢失内容。

#### Which issue(s) this PR fixes:
Fixes https://github.com/halo-dev/halo/issues/2773

#### Does this PR introduce a user-facing change?

```release-note
Console 端支持实时保存内容到浏览器，防止意外操作丢失内容。
```

### CODE REVIEW

File: src/modules/contents/posts/PostEditor.vue [hunk 1]
```diff
@@ -220,7 +222,15 @@ onMounted(async () => {
     // fetch post content
     await handleFetchContent();
   }
+  handleResetCache();
 });
+
+const { handleSetContentCache, handleResetCache, handleClearCache } =
+  useContentCache(
+    "post-catche",
```
Comment (ruibaby) [inline src/modules/contents/posts/PostEditor.vue:230]:
```suggestion
    "post-content-cache",
```

File: src/modules/contents/pages/SinglePageEditor.vue [hunk 2]
```diff
@@ -211,7 +214,15 @@ onMounted(async () => {
     // fetch single page content
     await handleFetchContent();
   }
+  handleResetCache();
 });
+
+const { handleSetContentCache, handleResetCache, handleClearCache } =
+  useContentCache(
+    "page-catche",
```
Comment (ruibaby) [inline src/modules/contents/pages/SinglePageEditor.vue:222]:
```suggestion
    "singlePage-content-cache",
```

**Review by ruibaby — CHANGES_REQUESTED** [review]

**Review by ruibaby — COMMENTED** [review]
/approve

实际测试符合预期。

**Review by ruibaby — APPROVED** [review]

**Review by guqing — COMMENTED** [review]
/lgtm

**Pull request merged**